### DISCUSSION TITLE
Querying an explicit many-to-many

### DISCUSSION THREAD

**Discussion by trongthienpc** [prisma/prisma#15241] [Q&A]
Hi everybody, i have following query:
 - Creates a new Post
 - Creates a new category assignment, or CategoriesOnPosts
 - Connects the category assignment to existing categories,  instead (with IDs 9 and 22), I don't know exactly input IDs. Example: [{Id: 1}, {Id: 2}, {Id: 3}, etc]. So how can we do that?
*** explicit many-to-many
```
const assignCategories = await prisma.post.create({
  data: {
    title: 'How to be Bob',
    categories: {
      create: [
        {
          assignedBy: 'Bob',
          assignedAt: new Date(),
          category: {
            connect: {
              id: 9, // ?
            },
          },
        },
        {
          assignedBy: 'Bob',
          assignedAt: new Date(),
          category: {
            connect: {
              id: 22, // ?
            },
          },
        },
      ],
    },
  },
})
```

**Comment by trongthienpc** [comment]
I have 1 solution for this case but it's difficult to scale:
```
listIds?.map(async (cat: categories) => {
          await prisma.postsOnCategories.create({
            data: {
              assignedBy: "admin",
              assignedAt: new Date(),
              categoryId: cat.Id,
              postId: post.Id,             
            },
          });
        });
```

**Reply by austincrim** [reply]
Hey @trongthienpc,

This is probably the solution I would go with. Why is it difficult to scale?

If you see a lot of performance issues, you could use [$queryRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#queryraw) instead.

**Reply by vladi-stevanovic** [reply]
👋 Hello @trongthienpc I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
